inference-engine/thirdparty/clDNN/src/graph_optimizer/reorder_inputs.cpp

```diff
@@ -52,7 +31,52 @@ void reorder_inputs::run(program& p) { run(p, _lo, _rf); }
 namespace {
 
 std::map<program_node*, format::type> get_preferred_formats(program& p, layout_optimizer& lo) {
+    GPU_DEBUG_GET_INSTANCE(debug_config);
+
     std::map<program_node*, format::type> fmt_map;
+
+#ifdef ENABLE_ONEDNN_FOR_GPU
+    size_t onednn_impls_counter = 0;
+    size_t all_impls_counter = 0;
+    float onednn_min_threshold = 0.1f;
+
+    // Calculate onednn kernels number and all kernels number inside the network
+    for (auto n : p.get_processing_order()) {
+        if (!n->is_in_data_flow())
+            continue;
+
+        auto ex = lo.get_preferred_format(*n);
+        auto impl = lo.get_preferred_impl_type(*n, ex);
+
+        if (impl == impl_types::onednn)
+            onednn_impls_counter++;
+
+        all_impls_counter++;
+    }
```

Comment:
Not sure how much overhead it causes, but now this is duplicated. What about calling it again only when we update `optimization_attributes_type::use_onednn_impls`?

Reply:
Fixed, please take a look at the updated variant.